src/app/clients/ClientsPage.tsx

```diff
@@ -85,56 +84,70 @@ const ClientsPage: React.FC<{}> = () => {
     const [filteredClientCount, setFilteredClientCount] = useState<number>(0);
     const [sortState, setSortState] = useState<SortState<ClientsTableRow>>({ sortEnabled: false });
     const [primaryFilters, setPrimaryFilters] = useState<Filter<ClientsTableRow, any>[]>(filters);
+    const clientTableFetchAbortController = useRef<AbortController | null>(null);
 
     const [perPage, setPerPage] = useState(10);
     const [currentPage, setCurrentPage] = useState(1);
     const startPoint = (currentPage - 1) * perPage;
     const endPoint = currentPage * perPage - 1;
 
     useEffect(() => {
-        const initialRequestParams: RequestParams<ClientsTableRow> = {
-            allFilters: { ...primaryFilters },
-            sortState: { ...sortState },
-            startPoint: startPoint,
-            endPoint: endPoint,
-        };
-        (async () => {
-            setIsLoading(true);
-            const filteredClientCount = await getClientsCount(supabase, primaryFilters);
-            const fetchedData = await getClientsData(
+        setIsLoading(true);
+        if (clientTableFetchAbortController.current) {
+            clientTableFetchAbortController.current.abort("stale request");
+        }
+        clientTableFetchAbortController.current = new AbortController();
+        if (clientTableFetchAbortController.current) {
+            getClientsCount(
+                supabase,
+                primaryFilters,
+                clientTableFetchAbortController.current.signal
+            )
+                .then((count) => setFilteredClientCount(count))
+                .catch();
```

Comment:
should we set error in these catches?

Reply:
I don't think so because it could be an aborted request, in which case we don't want to set an error

Reply:
What's stopping us from throwing a different custom error when the request is aborted, not the DatabaseError? Then in the catch, we can check if the thrown error is an instance of the DatabaseError or not

Reply:
sounds good! have done